### ISSUE TITLE
Syntax error or access violation: 1583 Incorrect parameters in the call to native function 'lower' 

### ISSUE THREAD

**Issue by suraj61018** [yajra/laravel-datatables#2292]
**I went through previous fixes but yet couldn't fix it**

Laravel version: 5.7
"yajra/laravel-datatables-oracle": "~8.0"

error: "Exception Message:↵↵SQLSTATE[42000]: Syntax error or access violation: 1583 Incorrect parameters in the call to native function 'lower' (SQL: select count(*) as aggregate from (select `acrm_contacts`.`id`, `acrm_contacts`.`company_id`, `acrm_contacts`.`group_id`, `acrm_contacts`.`phone1_code`, `acrm_contacts`.`phone1`, `acrm_contacts`.`phone2_code`, `acrm_contacts`.`phone2`, `acrm_contacts`.`email`, `acrm_contacts`.`skype`, `acrm_contacts`.`address`, `acrm_contacts`.`city`, `acrm_contacts`.`state_region`, `acrm_contacts`.`zip_postal_code`, `acrm_contacts`.`tax_id`, `acrm_contacts`.`country_id`, `acrm_contacts`.`thumbnail`, `acrm_contacts`.`name`, `acrm_contacts`.`fulladdress`, `acrm_contacts`.`is_user`, `acrm_contacts`.`follow_up`, `acrm_contacts`.`provience` from `acrm_contacts` where exists (select * from `acrm_roles` inner join `acrm_contact_contact_type` on `acrm_roles`.`id` = `acrm_contact_contact_type`.`contact_type_id` where `acrm_contacts`.`id` = `acrm_contact_contact_type`.`contact_id` and `id` != 11 order by `acrm_roles`.`id` desc) and (LOWER(`acrm_contacts`.`name`) LIKE %a% or exists (select * from `acrm_roles` inner join `acrm_contact_contact_type` on `acrm_roles`.`id` = `acrm_contact_contact_type`.`contact_type_id` where `acrm_contacts`.`id` = `acrm_contact_contact_type`.`contact_id` and LOWER(`acrm_roles`.`title`) LIKE %a% order by `acrm_roles`.`id` desc) or LOWER(`acrm_contacts`.`email`) LIKE %a% or LOWER(`acrm_contacts`.`address`) LIKE %a% or exists (select * from `acrm_contacts` as `acrm_laravel_reserved_0` inner join `acrm_contact_assigned_to_user` on `acrm_laravel_reserved_0`.`id` = `acrm_contact_assigned_to_user`.`user_id` where `acrm_contacts`.`id` = `acrm_contact_assigned_to_user`.`contact_id` and LOWER(`contacts` as `laravel_reserved_0.name`) LIKE %a% order by `acrm_laravel_reserved_0`.`id` desc)) and `acrm_contacts`.`deleted_at` is null order by `acrm_contacts`.`id` desc) count_row_table)"

my controller:
public function index( $type = '', $type_id = '' )
    {
        if (! Gate::allows('contact_access')) {
            return prepareBlockUserMessage();
        }

        if (request()->ajax()) {

            $query = Contact::query();

            $query->with("contact_type");

            $query->with("assigned_to");
            $template = 'actionsTemplate';

            if(request('show_deleted') == 1) {
                if (! Gate::allows('contact_delete')) {
                    return prepareBlockUserMessage();
                }
                $query->onlyTrashed();
                $template = 'restoreTemplate';
            }

            $query->select([
                'contacts.id',
                'contacts.company_id',
                'contacts.group_id',
                'contacts.phone1_code',
                'contacts.phone1',
                'contacts.phone2_code',
                'contacts.phone2',
                'contacts.email',
                'contacts.skype',
                'contacts.address',
                'contacts.city',
                'contacts.state_region',
                'contacts.zip_postal_code',
                'contacts.tax_id',
                'contacts.country_id',
                'contacts.thumbnail',

                'contacts.name',
                'contacts.fulladdress',
                'contacts.is_user',
                'contacts.follow_up',
                'contacts.provience',
            ]);
            if ( empty( $type ) ) {
                $query->whereHas("contact_type",
                    function ($query) {
                        $query->where('id', '!=', LEADS_TYPE);
                    });
            }

            /**
             * when we call invoices display from other pages!
             */
            if ( ! empty( $type ) && 'contact_type' === $type ) {
                $query->whereHas("contact_type",
                    function ($query) use( $type_id ) {
                        $query->where('roles.id', $type_id);
                    });
            }
            if ( ! empty( $type ) && 'contact_group' === $type ) {
                $query->when($type_id, function ($q, $type_id) {
                    return $q->where('contacts.group_id', $type_id);
                });
            }
            if ( ! empty( $type ) && 'contact_company' === $type ) {
                $query->when($type_id, function ($q, $type_id) {
                    return $q->where('contacts.company_id', $type_id);
                });
            }
            if ( ! empty( $type ) && 'country' === $type ) {
                $query->when($type_id, function ($q, $type_id) {
                    return $q->where('contacts.country_id', $type_id);
                });
            }
            if ( ! empty( $type ) && 'language' === $type ) {
                $query->when($type_id, function ($q, $type_id) {
                    return $q->where('contacts.language_id', $type_id);
                });
            }
            if ( ! empty( $type ) && 'department' === $type ) {
                $query->when($type_id, function ($q, $type_id) {
                    return $q->where('contacts.department_id', $type_id);
                });
            }

            $table = Datatables::of($query);
            $table->setRowAttr([
                'data-entry-id' => '{{$id}}',
            ]);
            $table->addColumn('massDelete', '&nbsp;');
            $table->addColumn('actions', '&nbsp;');
            $table->editColumn('actions', function ($row) use ($template) {
                $gateKey  = 'contact_';
                $routeKey = 'admin.contacts';

                return view($template, compact('row', 'gateKey', 'routeKey'));
            });

            $table->editColumn('thumbnail', function ($row) {
                if($row->thumbnail) { return '<a href="'. asset(env('UPLOAD_PATH').'/' . $row->thumbnail) .'" target="_blank"><img src="'. asset(env('UPLOAD_PATH').'/thumb/' . $row->thumbnail) .'"/>'; };
            });

            $table->editColumn('contact_type.title', function ($row) {
                if(count($row->contact_type) == 0) {
                    return '';
                } else {
                    return '<span class="label label-info label-many">' . implode('</span><span class="label label-info label-many"> ',
                            $row->contact_type->pluck('title')->toArray()) . '</span>';
                }

            });
            $table->editColumn('assigned_to.name', function ($row) {
                if(count($row->assigned_to) === 0) {
                    return '';
                }
                return '<span class="label label-info label-many">' . implode('</span><span class="label label-info label-many"> ',
                        $row->assigned_to->pluck('name')->toArray()) . '</span>';
            });
            $table->editColumn('first_name', function ($row) {
                return $row->name ? $row->first_name : '';
            });
            $table->editColumn('middle_name', function ($row) {
                return $row->middle_name ? $row->middle_name : '';
            });
            $table->editColumn('last_name', function ($row) {
                return $row->last_name ? $row->last_name : '';
            });
            $table->editColumn('contacts.name', function ($row) {
                $name = $row->name ? $row->name : '';
                return $name;
            });
            $table->editColumn('fulladdress', function ($row) {
                $fulladdress = $row->fulladdress ? $row->fulladdress : '';
                if ( empty( $fulladdress ) ) {
                    $fulladdress = $row->address ? $row->address : '';

                } else {
                    $fulladdress = str_replace("\n", ', ', $fulladdress);
                }
                return $fulladdress;
            });

            $table->editColumn('phone1', function ($row) {
                $code = $row->phone1_code ? '+' . $row->phone1_code . ' - ' : '';
                return $row->phone1 ? $code . $row->phone1 : '';
            });
            $table->editColumn('phone2', function ($row) {
                $code = $row->phone2_code ? '+' . $row->phone2_code . ' - ' : '';
                return $row->phone2 ? $code . $row->phone2 : '';
            });
            $table->editColumn('email', function ($row) {
                return $row->email ? $row->email : '';
            });
            $table->editColumn('address', function ($row) {
                return $row->address ? $row->address : '';
            });
            $table->editColumn('tax_id', function ($row) {
                return $row->tax_id ? $row->tax_id : '';
            });
            $table->editColumn('follow_up', function ($row) {
                return $row->follow_up ? $row->follow_up : '';
            });

            $table->rawColumns(['actions','massDelete','contact_type.title','assigned_to.name','language.language','follow_up']);
            return $table->make(true);
        }
        return view('admin.contacts.index', compact('type', 'type_id'));
    }

my script for viewing data on table:

$(document).ready(function () {
        @if ( ! empty( $type ) && ! empty( $type_id ) )
            window.dtDefaultOptionsNew.ajax.url = '{!! route('admin.list_contacts.index', [ 'type' => $type, 'type_id' => $type_id ] ) !!}?show_deleted={{ request('show_deleted') }}';
        @else
            window.dtDefaultOptionsNew.ajax.url = '{!! route('admin.contacts.index') !!}?show_deleted={{ request('show_deleted') }}';
        @endif
            window.dtDefaultOptionsNew.columns = [@can('contact_delete_multi')
            @if ( request('show_deleted') != 1 )
        {data: 'massDelete', name: 'id', searchable: false, sortable: false},
                @endif
                @endcan
            {data: 'contacts.name', name: 'contacts.name'},

            {data: 'contact_type.title', name: 'contact_type.title', sortable: false},

            {data: 'email', name: 'email'},
            {data: 'fulladdress', name: 'fulladdress'},
            {data: 'assigned_to.name', name: 'assigned_to.name',sortable: false},
            @if($type_id == 11)
            {data: 'follow_up', name: 'follow_up'},
            @endif
            {data: 'actions', name: 'actions', searchable: false, sortable: false}
        ];
        processAjaxTablesNew();
    });

**Comment by yajra** [comment]
It seems like you are using a self join relationship? It is currently not supported atm. Please use join query for the workaround.
